core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -330,6 +337,12 @@ default CompletableFuture<V> replaceAsync(K key, V value, Metadata metadata) {
       return replaceAsync(key, value, metadata.lifespan(), TimeUnit.MILLISECONDS, metadata.maxIdle(), TimeUnit.MILLISECONDS);
    }
 
+   default CompletionStage<CacheEntry<K, V>> replaceAsyncReturnEntry(K key, V value, Metadata metadata) {
+      // TODO: implement later
```

Comment:
I'll assume there is a JIRA for this :)

Reply:
@jabolina is working on the embedded versions of this, yes. The new API doesn't work until these are implemented. However, the old API works just fine. So I was just leaving them separate to not make this PR that big.

Reply:
Created #10207 for these TODOs.